File: packages/sms-mes-planning-pss-ui/src/types/GroupSequence.ts
import { Modify } from '..';

import { SequenceScenarioForm } from './SequenceScenario';

export interface GroupSequence {
  id: number;
  name: string;
  startDate: Date;
  remark: string | null;
  planningStatus: string;
  executionStatus: string | null;
  productionUnitGroupId: number;
  predecessorSequenceId: number | null;
  successorSequenceId: number | null;
}

export type GroupSequenceFormData = Modify<
  Omit<GroupSequence, 'planningStatus' | 'executionStatus'>,
  {
    id?: number;
    remark?: number;
    predecessorSequenceId?: number;
    successorSequenceId?: number;
    scenarioList: SequenceScenarioForm[];
  }
>;

File: packages/sms-mes-planning-pss-api/Models/GroupSequence.cs
using System;
using System.Collections.Generic;

namespace sms_mes_planning_pss_api.Models
{
    public class GroupSequence
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime StartDate { get; set; }
        public string Remark { get; set; }
        public string PlanningStatus { get; set; }
        public string ExecutionStatus { get; set; }
        public int ProductionUnitGroupId { get; set; }
        public int? PredecessorSequenceId { get; set; }
        public int? SuccessorSequenceId { get; set; }
        public IEnumerable<SequenceScenario> ScenarioList { get; set; }
    }
}
